### PR TITLE
Add a -c|--config flag to set a config path

### PR DESCRIPTION
**Pull request by aruhier**
Add a flag to set a custom config path.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -3,6 +3,7 @@
 #include "core/Hypridle.hpp"
 
 int main(int argc, char** argv, char** envp) {
+    std::string config_path = getMainConfigPath();
```
Comment (vaxerski) [inline src/main.cpp:6]: no_snake we use camelCase

File: src/config/ConfigManager.hpp [hunk 2]
```diff
@@ -7,9 +7,11 @@
 #include <vector>
 #include <memory>
 
+std::string getMainConfigPath();
```
Comment (vaxerski) [inline src/config/ConfigManager.hpp:10]: uhh, no. Make it static local in the configManager.cpp file and use it if configPath is empty.

File: src/main.cpp (hunk 1 above)
Comment (aruhier) [reply src/main.cpp]: Done, sorry for missing that.

File: src/config/ConfigManager.hpp (hunk 2 above)
Comment (aruhier) [reply src/config/ConfigManager.hpp]: I changed it to a ternary, tell me (or feel free to edit it directly) if you have a preference to something else.

File: src/main.cpp [hunk 3]
```diff
@@ -12,10 +13,15 @@ int main(int argc, char** argv, char** envp) {
 
         else if (arg == "--quiet" || arg == "-q")
             Debug::quiet = true;
+
+        else if (arg == "--config" || arg == "-c") {
+            configPath = argv[i + 1];
```
Comment (vaxerski) [inline src/main.cpp:18]: nit: you can do ++i here and omit the line below with i++

**Review by vaxerski — COMMENTED** [review]
rest lgtm

File: src/main.cpp (hunk 3 above)
Comment (aruhier) [reply src/main.cpp]: Done.

**Review by vaxerski — APPROVED** [review]
lgtm

**Pull request merged**